app/controllers/admin/provisioning_keys_controller.rb

```diff
@@ -0,0 +1,35 @@
+class Admin::ProvisioningKeysController < Admin::AdminController
+
+  def index
+    @provisioning_keys = ProvisioningKey.all
+  end
+
+  def show
+    @provisioning_key = ProvisioningKey.find(params[:id])
+  end
+
+  def new
+    @provisioning_key = ProvisioningKey.new
+  end
+
+  def create
+    @provisioning_key = ProvisioningKey.new(provisioning_key_params)
+    if @provisioning_key.save
+      redirect_to [:admin, @provisioning_key],
+        notice: "Provision key successfully created for environment #{@provisioning_key.api_env}"
+    else
+      render :new
+    end
+  end
+
+  def destroy
+    ProvisioningKey.find(params[:id]).destroy
```

Comment:
Change to `@provisioning_key.destroy`, see previous comment on setting the instance variable.